spec/controllers/api/v1/authentication_controller_spec.rb

```diff
@@ -41,6 +41,25 @@
       expect(parsed_body["errors"]).to eq("Mobile is invalid. Mobile can't be blank")
     end
 
+    context 'email' do
+      before do
+        set_browse_app_header
+      end
+      context 'when email is case-sensitive duplicate' do
+        it 'does not create a new user' do
+          user = create(:user)
+          expect { post :signup, format: 'json', user_auth: { mobile: '', email: user.email, first_name: '', last_name: '', address_attributes: { district_id: '', address_type: '' } } }.not_to change{ User.count }
```

Comment:
Can you also add a spec to check the validation returned is 'email is already taken'. That will confirm this spec is passing for the right reasons.

Reply:
this action has logic like 
- If user exists, ignore data and just send_verification_pin
- Otherwise, create new user and send pin

so it wont be throwing the error, instead it will just login if user already exists

Reply:
ok, thanks